singer_sdk/helpers/_compat.py

```diff
@@ -4,6 +4,8 @@
 
 import datetime
 import sys
+from importlib import metadata
+from typing import final  # noqa: ICN003
```

Comment:
We can get rid of these

Reply:
I will give it a try.  I removed them when I merged in SDK 0.35.0a3 alpha and pytest was not happy.

Reply:
Yep, pytest blew up again.  I will sync the changes so you can take a look.

Reply:
I think I see the issue now.  It is in io_base `SingerReader.listen` is using `@Final` and I was importing it from `_compact`

Reply:
I changed `@Final` to `@t.final` and `pytest` is happy 😃 now 🥳.